### PR TITLE
Add lambda embedded metrics plugin

### PR DESCRIPTION
**Pull request by mishaled**
## Which issues does it affect?

-  https://github.com/OS-Guild/monitored/issues/20

## What it does

This feature adds a support for [lambda EMF custom metrics ](https://docs.aws.amazon.com/lambda/latest/operatorguide/custom-metrics.html)

### CODE REVIEW

**Comment by korengal** [comment]
I think It will be easier to use this package:
https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/metrics/#getting-started

1. It writes to console in a format that AWS knows how to take it without async calls.
2. The api is easier
3. I would call the plugin CloudwatchPlugin

```        
const metrics = new Metrics({
            namespace: METRICS_NAMESPACE,
            serviceName: process.env.AWS_LAMBDA_FUNCTION_NAME || '',
            defaultDimensions: labels
        });

        metrics.addMetric(metricName, unit, value);
        metrics.publishStoredMetrics();
```

File: src/plugins/LambdaEmbeddedMetricsPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        throw new Error('Method not implemented.');
+    }
+
+    async timing(
+        name: string,
+        value: number,
+        tags?: Record<string, string> | undefined,
+        context?: Record<string, string> | undefined
+    ): Promise<void> {
+        await this.sendMetric(name, Unit.Seconds, value, tags, context, StorageResolution.High);
```
Comment (korengal) [inline src/plugins/LambdaEmbeddedMetricsPlugin.ts]: Are u sure its seconds?

**Review by korengal — CHANGES_REQUESTED** [review]
I think It will be easier to use this package:
https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/metrics/#getting-started

1. It writes to console in a format that AWS knows how to take it without async calls.
2. The api is easier



````
const metrics = new Metrics({
            namespace: METRICS_NAMESPACE,
            serviceName: process.env.AWS_LAMBDA_FUNCTION_NAME || '',
            defaultDimensions: labels
        });

        metrics.addMetric(metricName, unit, value);
        metrics.publishStoredMetrics();
````

In addition, I would call the plugin CloudwatchPlugin

File: src/plugins/LambdaEmbeddedMetricsPlugin.ts (hunk 1 above)
Comment (mishaled) [reply src/plugins/LambdaEmbeddedMetricsPlugin.ts]: @korengal You're right, its MS, I'll fix that

Comment (mishaled) [reply src/plugins/LambdaEmbeddedMetricsPlugin.ts]: Changed
[Here it does look like MS- https://github.com/mishaled/monitored-cloudwatch-plugin/blob/f825cd7278bdfe99ef50194742064933350b16f8/src/Monitor.ts#L62](https://github.com/OS-Guild/monitored/blob/11b1d6a6ae2b2e379c11109f0f9cda871f17e9b8/src/Monitor.ts#L62)

**Pull request closed without merging**